core/src/replay_stage.rs

```diff
@@ -329,6 +377,26 @@ impl ReplayStage {
 
         trace!("replay stage");
         // Start the replay stage loop
+
+        let (cost_update_sender, cost_update_receiver): (
+            Sender<ExecuteTimings>,
+            Receiver<ExecuteTimings>,
+        ) = channel();
+
+        let exit_clone = exit.clone();
+        let blockstore_clone = blockstore.clone();
+        let t_service = Builder::new()
+            .name("solana-replay-service".to_string())
+            .spawn(move || {
+                Self::service_loop(
+                    exit_clone,
+                    blockstore_clone,
+                    cost_model,
+                    cost_update_receiver,
+                );
+            })
+            .unwrap();
+
```

Comment:
In general, service specific logic should live in its own module to avoid bloating the already bloated replay module.

Good examples of this are services in `TransactionHistoryServices` like `CacheBlockMetaService` in `cache_block_meta_service.rs`, where only the sender needs to be passed into replay: https://github.com/solana-labs/solana/blob/master/core/src/validator.rs#L371-L378

Reply:
Thanks for pointing out. Another good standard to adhere. Making changes for it.